crates/uv/tests/tool_install.rs

```diff
@@ -380,7 +380,7 @@ fn tool_install_editable() {
         // We should have a tool receipt
         assert_snapshot!(fs_err::read_to_string(tool_dir.join("black").join("uv-receipt.toml")).unwrap(), @r###"
         [tool]
-        requirements = ["black @ file://[WORKSPACE]/scripts/packages/black_editable"]
+        requirements = [{ name = "black", directory = { install_path = "[WORKSPACE]/scripts/packages/black_editable", lock_path = "[WORKSPACE]/scripts/packages/black_editable", editable = true, url = { url = "file://[WORKSPACE]/scripts/packages/black_editable" } } }]
```

Comment:
If we're comfortable with the general idea here, I will iron out the actual requirement schema.